### ISSUE TITLE
unable to type render prop arguments for children

### ISSUE THREAD

**Issue by malbernaz** [developit/preact#1164]
```tsx
interface Foo {
  bar: string
} 

type Props = {
  children: (renderProps: Foo) => JSX.Element
}

class Parent extends Component<Props> { /* render props implementation */ }

const Child: FunctionalComponent = () => (
  <Parent>
    {/* not typed */}
    {({ bar }) => return <div />}
  </Parent>
)
```

For some reason children render arguments can't be typed. It works if implemented with another attribute (eg: `render`).

**Comment by developit** [comment]
@malbernaz `children` is an Array (regardless of how it is specified in JSX). Perhaps that's the issue?

**Comment by developit** [comment]
Hmm, not sure then. Maybe @marvinhagemeister has an idea.

**Comment by malbernaz** [comment]
I'm closing this here because the issue will be fixed in the next release.

**Comment by malbernaz** [comment]
Sorry for the mess. Reopening to leave as a suggestion to release a minor version updating the definition file (if it's not a breaking change).

**Comment by marvinhagemeister** [comment]
@malbernaz Sorry for the late response. This is fixed in master by #1116. We should definitely cut a new patch release sometime soon.

**Comment by developit** [comment]
This should be fixed in [8.3.0](https://github.com/developit/preact/releases/tag/8.3.0)!
Please feel free to re-open otherwise :)
